crates/swc/src/plugin.rs

```diff
@@ -14,10 +14,61 @@ use swc_ecma_visit::{noop_fold_type, Fold};
 #[serde(deny_unknown_fields, rename_all = "camelCase")]
 pub struct PluginConfig(String, serde_json::Value);
 
-pub fn plugins(names: Option<Vec<PluginConfig>>) -> impl Fold {
+struct PluginBinary(String, Vec<u8>, String);
+
+pub fn plugins(config: crate::config::JscExperimental) -> impl Fold {
     #[cfg(feature = "plugin")]
     {
-        RustPlugins { plugins: names }
+        use anyhow::Context;
+
+        let mut plugin_binaries = vec![];
+        let mut plugin_errors = vec![];
+
+        if let Some(plugins) = &config.plugins {
```

Comment:
Somewhat ugly, but thoughts? Peeking paths I realized there are 2 things occurring per each process for 1. Config for plugin 2. Reading file itself.

Instead of trying to cache both individually inside of runner, I put those upfront and pass into each plugin invocation. This'll gets rid of internal cache mangement and makes cache lookup to plugin contents (with config) to zero, but as you could see in code error handling gets little dirty due to lacks of my knowledge.

@kdy1

Reply:
Seems much better than the previous one to me.